export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+  Returns:
+    int, timestamp adjusted to multiple of step.
+  """
+  return timestamp - (timestamp % step)
+
+
+def default_end_time(options):
+  """Calculates a default, end timestamp.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.step, options.update, options.ts_offset.
+  Returns:
+    int, timestamp in seconds since the epoch.
+  """
+  ts_current = int(time.time())
+
+  # Align current ts to a multiple of length.
+  ts_end = align_timestamp(ts_current, options.length)
+
+  # Align end ts to a multiple of step size (just in case).
+  ts_end = align_timestamp(ts_end, options.step)
+
+  if options.update and (ts_current - ts_end) < options.ts_offset:
+    raise TimeOptionError('ts_current - ts_end < min offset: %s - %s < %s' %
+        (ts_current, ts_end, options.ts_offset))
+  return ts_end
+
+
+def default_start_time(options, ts_previous):
+  """Calculates a default start timestamp.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.step, options.update, options.ts_end.
+    ts_previous: int, timestamp in seconds since epoch of last
+        successful export. On first run, this value should be zero.
+  Returns:
+    int, timestamp in seconds since the epoch.
+  """
+  # Only when both options.update and ts_previous are set, skip assignment.
+  if not (options.update and ts_previous):
+    # If either update or ts_previous were not set, then choose a default.
+    ts_previous = options.ts_end - options.length
+
+  return align_timestamp(ts_previous, options.step)
+
+
+def assert_start_and_end_times(options):
+  """Performs a sanity check on start and end timestamps.
+
+  This method asserts that both ts_end is less than ts_start and that the
+  difference between them is greater than options.length.
+
+  Args:
+    options: flags.FlagValues, the runtime options. These values are read:
+        options.length, options.ts_start, options.ts_end.
+  Raises:
+    TimeOptionError, if a start & end time constraint is violated.
+  """
+  # Always check if basic constratins are respected.
+  if options.ts_end < options.ts_start:
+    raise TimeOptionError('Start time cannot be later than end time.')
+
+  if options.ts_end - options.ts_start < options.length:
+    msg = ('Start and end times difference is less than length: '+
+           '%s - %s = %s < %s' % (options.ts_end, options.ts_start,
+           (options.ts_end - options.ts_start), options.length))
+    raise TimeOptionError(msg)
+
+  logging.debug('Exporting: %s to %s', time.ctime(options.ts_start),
+                time.ctime(options.ts_end))
+
+
+def default_output_name(ts_start, ts_end, output_dir):
+  """Creates a default output filename based on time range and output dir.
+
+  Filenames are formatted with time stamps as:
+      <output_dir>/resource-utilization/YYYY/MM/DD/<HOSTNAME>/
+          metrics-YYYYMMDDTHHMMSS-to-YYYYMMDDTHHMMSS.json
```

Comment:
Can we specify whether the directory `YYYY/MM/DD` is based on the `ts_start` or `ts_end`?